link_linux.go

```diff
@@ -1580,9 +1586,57 @@ func LinkDeserialize(hdr *unix.NlMsghdr, m []byte) (Link, error) {
 	}
 	*link.Attrs() = base
 
+	// If the tuntap attributes are not updated by netlink due to
+	// an older driver, use sysfs
+	if link != nil && linkType == "tun" {
+		tuntap := link.(*Tuntap)
+
+		if tuntap.Mode == 0 {
+			ifname := tuntap.Attrs().Name
+			if flags, err := readSysPropAsInt64(ifname, "tun_flags"); err == nil {
```

Comment:
Why aren't we handling the error ?

Reply:
@aboch I chose to ignore this error, in case sysfs has not been mounted for the network namespace, resulting in interfaces not found on /sys.
I took the approach of best-effort to get the interface attributes. If you prefer erroring out here, 
I can follow that approach.

Reply:
>I took the approach of best-effort to get the interface attributes.

Makes sense. Thanks.

Reply:
Thanks @aboch. I have addressed your other comments. PTAL.